perfeed/data_stores/main.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any
+from pydantic import BaseModel
+from perfeed.data_stores.base import (
+    BaseStorage, 
+    UnsupportedFormatError
+)
+from perfeed.data_stores.storage_feather import FeatherStorage
+from perfeed.data_stores.storage_sqldb import SQLStorage
+from perfeed.models.pr_summary import PRSummary
+import pandas as pd
+from typing import Dict
+
+
+class DataStore:
+    """Storage frontend handling multiple storage backends."""
+    
+    def __init__(self, storage: BaseStorage=FeatherStorage) -> None:
+        self.storage = storage
+
+    def save(self, data: BaseModel, metadata: BaseModel) -> None:
+        self.storage.save(data=data, metadata=metadata)
+
+    def load(self) -> pd.DataFrame:
+        return self.storage.load()
+
+    @classmethod
+    def from_format(cls, store_type: str, **kwargs) -> BaseStorage:
```

Comment:
I think we can save this factory function and just let the user to create the individual storage instance. After all, the user still pass the whole parameters with or without the factory anyway.

with factory:
```
ds2 = DataStore.from_format(    
        # store_type="feather",
        store_type="sql",
        data_type="pr_summary",
        overwrite=True, 
        append=False
    )
```

without factory:
```
FeatherStorage(data_type="pr_summary", overwrite=True, append=False)
```

Reply:
the idea is that it would be easier to remember `store_type = feather` than to call `FeatherStorage`  from package, which u have to look up the document. i am up to either.

Reply:
I'd suggest removing this factory so we won't have to keep updating the `if-else` as we support more store types
```
        if store_type == "feather":
            return cls(FeatherStorage(data_type=data_type, overwrite=overwrite, append=append))

        elif store_type == "sql":            
            return cls(SQLStorage(data_type=data_type, overwrite=overwrite, append=append))
        else:
            raise UnsupportedFormatError(f"{store_type} is not supported.")
```